runtime/src/main/java/org/corfudb/util/Utils.java

```diff
@@ -228,10 +204,32 @@ public static long getLogTail(Layout layout, CorfuRuntime runtime) {
         } else if (segment.getReplicationMode() == Layout.ReplicationMode.QUORUM_REPLICATION) {
             throw new UnsupportedOperationException();
         }
-
         return globalLogTail;
     }
 
+    public static long getLogSize(Layout layout, CorfuRuntime runtime, long startAddress, long endAddress) {
+        long size = 0;
+        long segmentSize = 0;
+        long currentEnd = 0;
+        for (Layout.LayoutSegment segment : layout.getSegments()) {
+            if (segment.getEnd() > 0 && segment.getEnd() <= startAddress || endAddress <= segment.getStart())
```

Comment:
Can we add some comments. So reader can quickly understand that this logic is looking for the correct log unit to query.